src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -222,6 +261,23 @@ protected override bool Receive(object message)
 
                     if (_stageActorsMap.GetOrElse(requestMessages.Topics, Sender).Equals(Sender))
                         _requests = _requests.SetItem(Sender, requestMessages);
+                    
+                    /* UNREQUESTED MESSAGES CHECK */
+                    if (_unRequestedMessages.Count > 0)
```

Comment:
If we have unrequested messages that match the requestor's requirements, deliver them to the requesting stage immediately and skip the poll until the next request comes in.